### ISSUE TITLE
Asynchronous execution of sudo in parallel breaks terminal

### ISSUE THREAD

**Issue by sylv-io** [sudo-project/sudo#312]
Regarding the issue addressed here: https://github.com/go-task/task/issues/1369, it looks like there is a regression that causes the tty to break when sudo is running in parallel.
As mentioned in the above issue, it does not require a password prompt to break and can be reproduced by this simple go application:
 
 ```
 package main

import (
        "log"
        "os"
        "os/exec"
        "sync"
)

const NumExecutions = 3

func main() {
        var wg sync.WaitGroup

        for i := 0; i < NumExecutions; i++ {
                wg.Add(1)

                go func() {
                        defer wg.Done()

                        cmd := exec.Command("sudo", "true")

                        cmd.Stdin = os.Stdin
                        cmd.Stdout = os.Stdout
                        cmd.Stderr = os.Stderr

                        err := cmd.Run()
                        if err != nil {
                                log.Fatal(err)
                        }
                }()
        }

        wg.Wait()
}
```

I'm pretty sure this didn't happen about 1 year ago, so I'm currently trying to bisect this bug.

**Comment by sylv-io** [comment]
I can confirm that this bug does not occur with sudo v1.9.0.

**Comment by millert** [comment]
Can you check whether setting

Defaults !use_pty

in the sudoers file works around the problem?

**Comment by sylv-io** [comment]
> Can you check whether setting
> 
> Defaults !use_pty
> 
> in the sudoers file works around the problem?

hi @millert
You are correct. This is indeed a workaround for the problem. :+1:

**Comment by sylv-io** [comment]
I git bisected and as expected, this is the breaking commit:
`Enable the use_pty option by default for sudo 1.9.14.` 894daa88f66a69d3c6fe4d2255183d4c0b33a464

**Comment by millert** [comment]
Fixed by [fabb626](https://github.com/sudo-project/sudo/commit/fabb6264fc3a001fa1d182f3663370fa28bccbbb)

**Comment by sylv-io** [comment]
Thanks :+1:
